src/combinatorics/simple_graph/basic.lean

```diff
@@ -640,6 +640,30 @@ lemma delete_edges_eq_inter_edge_set (s : set (sym2 V)) :
   G.delete_edges s = G.delete_edges (s ∩ G.edge_set) :=
 by { ext, simp [imp_false] { contextual := tt } }
 
+/-! ## Induced graphs -/
+
+/- Given a set `s` of vertices, we can restrict a graph to those vertices by restricting its
+adjacency relation. This gives a map between `simple_graph V` and `simple_graph s`.
+
+There is also a notion of induced subgraphs (see `simple_graph.subgraph.induce`). -/
+
+/-- Restrict a graph to the vertices in the set `s`. -/
+@[simps]
+def induce (s : set V) (G : simple_graph V) : simple_graph s :=
```

Comment:
Should we rather take a predicate `p : V → Prop`? This is morally equivalent but more general for formalization purposes.

Also I see where the name is coming from but it does sound very mathlib.